sdk/core/core/internal/az_config_internal.h

```diff
@@ -12,7 +12,7 @@
 #include <_az_cfg_prefix.h>
 enum
 {
-  AZ_HTTP_REQUEST_HEADER_BUF_SIZE = AZ_HTTP_REQUEST_HEADER_COUNT * sizeof(az_pair),
+  _az_HTTP_REQUEST_HEADER_BUF_SIZE = AZ_HTTP_REQUEST_HEADER_COUNT * sizeof(az_pair),
```

Comment:
I think AZ_HTTP_REQUEST_HEADER_COUNT should not be defined at all in core. The number of headers required really depends on the service and, in fact, on a specific azure service operation. So, this is better defined by the client library code (not in core) as only the client library code knows what the number of headers it needs.

Reply:
Let's change it to per-client then, during our meeting later today.

Reply:
Done :)